Interface/BlackMageHudWindow.cs

```diff
@@ -0,0 +1,190 @@
+                cursorPos.X = cursorPos.X + barSize.X + 3;
+            }
+        }
+
+        protected virtual void DrawPolyglot()
+        {
+            var gauge = PluginInterface.ClientState.JobGauges.Get<BLMGauge>();
+            var barSize = new Vector2(18, 12);
+            var scale = gauge.IsEnoActive() ? (gauge.NumPolyglotStacks == 2 ? 1 : gauge.TimeUntilNextPolyglot / 30000f) : 1;
+            scale = 1 - scale;
+
+            var drawList = ImGui.GetWindowDrawList();
+
+            if (gauge.NumPolyglotStacks == 0)
+            {
+                var cursorPos = new Vector2(CenterX - barSize.X / 2f, CenterY + YOffset - 85);
+                DrawPolyglotStack(cursorPos, barSize, scale, false);
+            } 
+            else
+            {
+                // 1st stack (charged)
+                var cursorPos = new Vector2(CenterX - barSize.X - 1, CenterY + YOffset - 85);
+                DrawPolyglotStack(cursorPos, barSize, 1, true);
+
+                // 2nd stack
+                bool is2ndCharged = gauge.NumPolyglotStacks == 2;
+                cursorPos.X = CenterX + 1;
+                DrawPolyglotStack(cursorPos, barSize, is2ndCharged ? 1 : scale, is2ndCharged);
+            }
+        }
+
+        private void DrawPolyglotStack(Vector2 position, Vector2 size, float scale, bool isFullyCharged)
+        {
+            var drawList = ImGui.GetWindowDrawList();
+
+            // glow
+            if (isFullyCharged)
+            {
+                var glowPosition = new Vector2(position.X - 1, position.Y - 1);
+                var glowSize = new Vector2(size.X + 2, size.Y + 2);
+                drawList.AddRectFilled(glowPosition, glowPosition + glowSize, 0xAAFFFFFF);
+            }
+
+            // background
+            drawList.AddRectFilled(position, position + size, 0x88000000);
+
+            // fill
+            if (isFullyCharged)
+            {
+                drawList.AddRectFilledMultiColor(position, position + size, 0xFF5F3A93, 0xFFBE8FFF, 0xFFBE8FFF, 0xFF5F3A93);
+            }
+            else
+            {
+                drawList.AddRectFilled(position, position + new Vector2(size.X * scale, size.Y), 0xFF5F3A93);
+            }
+
+            // black border
+            drawList.AddRect(position, position + size, 0xFF000000);
+        }
+
+        protected virtual void DrawTripleCast()
+        {
+            var tripleStackBuff = PluginInterface.ClientState.LocalPlayer.StatusEffects.FirstOrDefault(o => o.EffectId == 1211);
+            var drawList = ImGui.GetWindowDrawList();
+
+            var positions = new ArrayList();
+            positions.Add(new Vector2(CenterX, CenterY + YOffset - 100));
+            positions.Add(new Vector2(CenterX - JobBarWidth / 2f + 30, CenterY + YOffset - 85));
+            positions.Add(new Vector2(CenterX + JobBarWidth / 2f - 30, CenterY + YOffset - 85));
+
+            for (int i = 0; i < 3; i++)
+            {
+                if (tripleStackBuff.StackCount <= i)
```

Comment:
Can just loop using the StackCount as a range.

```suggestion
               for (int i= 0; i< TripleStackBuff.StackCount; i++)
               {
                   drawList.AddCircle((Vector2)positions[i], 10, 0xFFFFFFFF, 6, 4);
               }
```